content/blog/client_node_and_cncf_mentorship.md

```diff
@@ -0,0 +1,208 @@
+
+## My thoughts
+
+Alright that's enough of technical talk and now I want to briefly share my experience on the mentorship program. My early career at Microsoft focused on developing proprietary products. This is really the first time that I've contributed to a large open source and community maintained project. Because of this, I was able to collaborate with engineers outside of my employer and share insights in the field. Calling out to my great mentor [Andy Lok](https://github.com/andylokandy), who had helped me a lot in early designing and unblocking work items, and we had great talks about distributed systems, programming languages and coding. I was even more inspired to learn that he is a core maintainer for Idris, which is a cool dependently typed language and hopefully I will learn it one day. 
+
+Back to the open source development experience. Compare with my career, there are a lot differences. For example, It is unimaginable for me to develop a completely new project on my own pace when I participated the CNCF TiKV mentorship. The amount of flexibility on project management, choosing the tools and communication with my mentor is unparallel. I've gone through a complete end-to-end project lifecycle, from designing the client, to choosing the right technologies, to the real implementation and finding ways to unblock myself, and finally to the release of the client to npm. More importantly, I own my work. Near the end of my mentorship journey, I told my Andy that I wanted to be a long-term contributor to TiKV. That is, I want to keep updating the NodeJS client with new features, track GitHub issues and review pull requests, and potentially even increase my scope to other TiKV clients. This sense of responsibility and attachment naturally comes from my initial efforts that establish this new project, and surely it trained me to be a better developer, and a more confident contributor to open source projects. At the end, TiKV community rewards my effort by [promoting](https://github.com/tikv/community/pull/148) me to become a official reviewer for the project. 
+
+One specific challenge around open source projects, and I struggled a lot is losing track of progress or procrastination in general. This is not a new topic and a lot time management researches and books pointed out methods and practices to become better at it. Personally, I found discipline and consistency are the most important traits to become a good open source contributor, due to the fact that there is no tight deadlines or people who pushed me to complete work items. I tried and failed many times to at least code or learn something at least half an hour every day. It is just way easier to give up on the work and becomes surprisingly difficult to pick it up later. So keep this in mind: in open source community, you will be fighting procrastination much more often than your regular work and you must be motivated and passionate at the things you do.
```

Comment:
Great insight! I personally found that time management in open source projects works lots differently from the company internal work. And worse, the ideas coming from project management books start to conflict with each other when a task is driven by community and business together. I've been reading a few new books about maintaining open-source communities, and I hope to share them with you next time.

Reply:
Sure! Let's find a time to connect